File: leetcode/283MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int curr_size = 0;
        for(int i=0;i<n;i++){
            if(nums[i] != 0){
                int temp = nums[i];
                nums[i] = nums[curr_size];
                nums[curr_size] = temp;
                curr_size++;
            }
        }
    }
}

File: leetcode/283MoveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        first_zero = -1
        for i,n in enumerate(nums):
            if n == 0 and first_zero == -1:
                first_zero = i
            if n != 0 and first_zero != -1:
                nums[i], nums[first_zero] = nums[first_zero], nums[i]
                first_zero += 1
